skynet/modules/stt/streaming_whisper/state.py

```diff
@@ -136,37 +156,44 @@ def add_to_store(self, chunk: bytes):
             log.debug(f'Participant {self.participant_id}: the chunk is silent.')
             self.silent_chunks += 1
         else:
+            if self.working_audio_starts_at == 0:
+                self.working_audio_starts_at = self.received_timestamp_from_chunk - int(self.chunk_duration * 1000)
             self.long_silence = False
             self.silent_chunks = 0
 
-    def trim_working_audio(self, bytes_to_cut: int):
+    def trim_working_audio(self, bytes_to_cut: int) -> bytes:
         log.debug(
             f'Participant {self.participant_id}: '
             + f'trimming the audio buffer, current length is {len(self.working_audio)} bytes.'
         )
-        self.working_audio = bytearray(self.get_num_bytes_for_slicing(1.0)) + self.working_audio[bytes_to_cut:]
+        dropped_chunk = self.working_audio[:bytes_to_cut]
+        self.working_audio = self.working_audio[bytes_to_cut:]
         log.debug(
             f'Participant {self.participant_id}: '
             + f'the audio buffer after cut is now {len(self.working_audio)} bytes'
         )
+        return dropped_chunk
 
-    def get_response_payload(self, transcription: str, final: bool = False) -> utils.TranscriptionResponse:
+    def get_response_payload(
+        self, transcription: str, start_timestamp: int, final_audio: bytes | None = None, final: bool = False
+    ) -> utils.TranscriptionResponse:
         return utils.TranscriptionResponse(
-            id=self.transcription_id,
+            id=str(utils.uuid_from_time(start_timestamp)),
             participant_id=self.participant_id,
-            ts=self.transcription_ts if final else utils.now(),
+            ts=start_timestamp,
             text=transcription,
+            audio=base64.b64encode(final_audio).decode('ASCII') if final_audio else '',
```

Comment:
Can we skip this if the log level is not debug, or maybe even another env variable? I can imagive privacy concers, but also performance.

Reply:
I already defined an env var which is off by default